src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1369,6 +1499,58 @@ private void EmitSwitchDispatch(BoundSwitchDispatch dispatch)
             }
         }
 
+#nullable enable
+        private Cci.IMethodReference? GetLengthMethodRef(SyntaxNode syntaxNode, TypeSymbol keyType, bool isReadOnlySpan, bool isSpanOrReadOnlySpan)
+        {
+            if (isSpanOrReadOnlySpan)
+            {
+                var spanTLengthMethod = (MethodSymbol)Binder.GetWellKnownTypeMember(_module.Compilation,
+                    (isReadOnlySpan ? WellKnownMember.System_ReadOnlySpan_T__get_Length : WellKnownMember.System_Span_T__get_Length),
+                    _diagnostics, syntax: syntaxNode);
+
+                Debug.Assert(spanTLengthMethod != null && !spanTLengthMethod.HasUseSiteError);
```

Comment:
>Debug.Assert(spanTLengthMethod != null && !spanTLengthMethod.HasUseSiteError) [](http://example.com/codeflow?start=16&length=77)

Why are we able to assert here but not for the indexer below?

Reply:
I'm not completely sure. This was existing logic that I just factored out.
For the new scenario where I use this helper (`GetLengthMethodRef`), I assert that the return value is not null (ie. we always get length methods for spans or strings).